dll/win32/shell32/wine/shell32_main.c

```diff
@@ -310,6 +310,28 @@ LPWSTR* WINAPI CommandLineToArgvW(LPCWSTR lpCmdline, int* numargs)
     return argv;
 }
 
+static HRESULT SHELL_GetDetailsOfToBuffer(IShellFolder *psf, PCUITEMID_CHILD pidl,
+                                          UINT col, LPWSTR Buf, UINT cchBuf)
+{
+    IShellFolder2 *psf2;
+    IShellDetails *psd;
+    SHELLDETAILS details;
+    HRESULT hr = IShellFolder_QueryInterface(psf, &IID_IShellFolder2, (void**)&psf2);
+    if (SUCCEEDED(hr))
+    {
+        hr = IShellFolder2_GetDetailsOf(psf2, pidl, col, &details);
+        IShellFolder2_Release(psf2);
+    }
+    else if (SUCCEEDED(hr = IShellFolder_QueryInterface(psf, &IID_IShellDetails, (void**)&psd)))
```

Comment:
These QueryInterface can also use the `IID_PPV_ARG(...)`

Reply:
One would think so but it does not work.

Reply:
Doing COM in c makes no sense anyway, so I am not surprised that this macro got never fixed.